Sources/Markdown/Walker/Walkers/MarkupFormatter.swift

```diff
@@ -958,9 +980,21 @@ public struct MarkupFormatter: MarkupWalker {
         /// extending each line with spaces to fit the uniform column width.
         let expandedHeaderCellTexts = (0..<uniformColumnCount)
             .map { column -> String in
-                let minLineLength = finalColumnWidths[column]
-                return headCellTexts[column]
-                    .ensuringCount(atLeast: minLineLength, filler: " ")
+                let colspan = headCellSpans[column]
+                if colspan == 0 {
+                    // If this cell is being spanned over, collapse it so it
+                    // can be filled with the spanning cell.
+                    return ""
+                } else {
+                    // With a colspan, we want to expand the cell width to
+                    // cover multiple columns. This helpfully generalizes
+                    // to `colspan == 1`, where it will only query the
+                    // current column!
+                    let lastColumn = column + Int(colspan)
+                    let minLineLength = (column..<lastColumn).map({ finalColumnWidths[$0] }).reduce(0, { $0 + $1 })
```

Comment:
Should this logic be refactored?

Reply:
Good idea! Just pushed up a new commit combining the logic.